### PR TITLE
ME-1788 Fix Sudo behaviour with border0 cli

### PR DESCRIPTION
**Pull request by atoonk**
# Description

This change implements some of the user feedback regarding the VPN log messages and  HTTP proxy crash.
Also making sure the user can use sudo without losing his or her preferences, ie we use the original users preference, admin and client token locations.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: cmd/login.go [hunk 1]
```diff
@@ -79,7 +81,21 @@ var loginCmd = &cobra.Command{
...
+				sudoAttempt := false
+				if runtime.GOOS == "darwin" && sudoUsername != "" {
+					err = exec.Command("sudo", "-u", sudoUsername, "open", url).Run()
+					if err == nil {
+						// If for some reason this failed, we'll try again to old way
+						sudoAttempt = true
+					}
+				}
+				if !sudoAttempt {
+					_ = open.Run(url)
```
Comment (adrianosela) [inline cmd/login.go:97]: no need to catch return value here. Though consider printing the error.

File: internal/client/resources.go [hunk 2]
```diff
@@ -96,7 +98,21 @@ func Login(org string) (token string, claims jwt.MapClaims, err error) {
...
-	_ = open.Run(url)
+
+	// check if we're on DARWIN and if we're running as sudo, if so, make sure we open the browser as the user
+	// this prevents folsk from not having access to credentials , sessions, etc
+	sudoUsername := os.Getenv("SUDO_USER")
+	sudoAttempt := false
+	if runtime.GOOS == "darwin" && sudoUsername != "" {
+		err = exec.Command("sudo", "-u", sudoUsername, "open", url).Run()
+		if err == nil {
+			// If for some reason this failed, we'll try again to old way
```
Comment (adrianosela) [inline internal/client/resources.go:109]: This is confusing -- you mention "if for some reason this failed" but you are adding that in the `err == nil` clause?

**Review by adrianosela — APPROVED** [review]
LGTM but we should test this on windows and linux machines before we merge it.

File: internal/client/resources.go (hunk 2 above)
Comment (atoonk) [reply internal/client/resources.go:109]: arreed will fix

File: cmd/login.go (hunk 1 above)
Comment (atoonk) [reply cmd/login.go:97]: not printing the error by design, it may fail, we wanna fail silently. This is the same as before

**Pull request merged**